qiita_pet/handlers/study_handlers/listing_handlers.py

```diff
@@ -42,14 +43,46 @@ def _get_shared_links_for_study(study):
     return ", ".join(shared)
 
 
-def _build_study_info(user, results=None):
-    """builds list of dicts for studies table, with all html formatted"""
+def _build_study_info(user, study_proc=None, proc_samples=None):
+    """builds list of dicts for studies table, with all html formatted
+
+    Parameters
+    ----------
+    user : User object
+        logged in user
+    study_proc : dict of lists, optional
+        Dictionary keyed on study_id that lists all processed data associated
+        with that study. Required if proc_samples given.
+    proc_samples : dict of lists, optional
+        Dictionary keyed on proc_data_id that lists all samples associated with
+        that processed data. Rquired if study_proc given.
+
+    Returns
+    -------
+    infolist: list of dict of lists and dicts
```

Comment:
I don't know, probably worth give it a shot and try different things. The code as it is right now is really hard to follow and I don't think that is maintainable.
On the other hand, are you forced to use JSON? The reason why I'm asking is because it looks like you're using all this information in a javascript function, so you probably can just pass the python object and use the template engine to fill those values.